app/models/course/assessment/question/text_response.rb

```diff
@@ -7,10 +7,19 @@ class Course::Assessment::Question::TextResponse < ApplicationRecord
   has_many :solutions, class_name: Course::Assessment::Question::TextResponseSolution.name,
                        dependent: :destroy, foreign_key: :question_id, inverse_of: :question
 
+  has_many :groups, class_name: Course::Assessment::Question::TextResponseComprehensionGroup.name,
+                    dependent: :destroy, foreign_key: :question_id, inverse_of: :question
+
   accepts_nested_attributes_for :solutions, allow_destroy: true
 
+  accepts_nested_attributes_for :groups, allow_destroy: true
+
   def auto_gradable?
-    !solutions.empty?
+    if comprehension_question?
+      groups.map(&:auto_gradable_group?).any?
```

Comment:
Should it be `auto_gradable?` only if all groups are auto_gradable?

Reply:
This has implications because we only allow autogradable questions to be added to autograded Assessments.

Reply:
I'd beg to differ. So long as there is *a* Solution, I'd call it as an autogradable question.
`any?` so that if there are any Groups with no Points, or any Points with no Solutions, it is still fine.

Reply:
Ok ignore this, my misunderstanding.

Reply:
Would `groups.any?(&:auto_gradable_group?)` be better?

Reply:
It does sound better and clearer. Will change it.

Reply:
If I'm not mistaken, `any?` will return once the first `&:auto_gradable_group?` evaluates to `true`, whereas we are guaranteed to be evaluating for all groups if we used `map`.

Remember to change for the other compre models as well!

Reply:
Changed in #2790.